src/boot/ksdk1.1.0/devRV8803C7.c

```diff
@@ -0,0 +1,343 @@
+	*	Redistributions of source code must retain the above
+		copyright notice, this list of conditions and the following
+		disclaimer.
+
+	*	Redistributions in binary form must reproduce the above
+		copyright notice, this list of conditions and the following
+		disclaimer in the documentation and/or other materials
+		provided with the distribution.
+
+	*	Neither the name of the author nor the names of its
+		contributors may be used to endorse or promote products
+		derived from this software without specific prior written
+		permission.
+
+	THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+	"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+	LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+	FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+	INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+	BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+	LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+	CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+	LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+	ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include <stdint.h>
+#include "fsl_rtc_driver.h"
+#include "fsl_i2c_master_driver.h"
+#include "fsl_spi_master_driver.h" /* Crazy we need to include this as spi is unused, but otherwise we need to put it in warp.h, which is against the coding style. */
+
+#include "warp.h"
+#include "devRV8803C7.h"
+
+
+typedef enum {kWarpRV8803RegSec = 0x00,
+              kWarpRV8803RegMin = 0x01,
+              kWarpRV8803RegHour = 0x02,
+              kWarpRV8803RegWeekday = 0x03,
+              kWarpRV8803RegDate = 0x04,
+              kWarpRV8803RegMonth = 0x05,
+              kWarpRV8803RegYear = 0x06,
+              kWarpRV8803RegRAM = 0x07,
+              kWarpRV8803RegMinAlarm = 0x08,
+              kWarpRV8803RegHourAlarm = 0x09,
+              kWarpRV8803RegWeekdayOrDateAlarm = 0x0A,
+              kWarpRV8803RegTimerCounter0 = 0x0B,
+              kWarpRV8803RegTimerCounter1 = 0x0C,
+              kWarpRV8803RegExt = 0x0D,
+              kWarpRV8803RegFlag = 0x0E,
+              kWarpRV8803RegCtrl = 0x0F,
+} kWarpRV8803Reg;
+
+#define BIT(n) (uint8_t)1U << n
+
+typedef enum {kWarpRV8803ExtClrTD = 0xFC,
+              kWarpRV8803ExtClrFD = 0xF3,
+} kWarpRV8803ExtClr;
+
+typedef enum {kWarpRV8803ExtTE = BIT(4),
+              kWarpRV8803ExtUSEL = BIT(5),
+              kWarpRV8803ExtWADA = BIT(6),
+              kWarpRV8803ExtTEST = BIT(7),
+} kWarpRV8803ExtFlag;
+
+typedef enum {kWarpRV8803FlagV1F = BIT(0),
+              kWarpRV8803FlagV2F = BIT(1),
+              kWarpRV8803FlagEVF = BIT(2),
+              kWarpRV8803FlagAF = BIT(3),
+              kWarpRV8803FlagTF = BIT(4),
+              kWarpRV8803FlagUF = BIT(5),
+} kWarpRV8803FlagFlag;
```

Comment:
is `FlagFlag` what you intended?

Reply:
Yes, it did seem weird but I couldn't thing of a good name for the flags in the flag register. I wanted to maintain consistency with kWarpRV8803CtrlFlag and kWarpRV8803ExtFlag